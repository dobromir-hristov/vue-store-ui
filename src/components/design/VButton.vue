<template>
  <button
    :class="[`v-button--${color}`, { 'is-outlined': outlined }]"
    class="v-button"
  >
    <slot/>
  </button>
</template>

<script>
/**
 * @module Button
 */
export default {
  name: 'VButton',
  props: {
    color: {
      type: String,
      default: 'primary'
    },
    outlined: {
      type: Boolean,
      default: false
    }
  }
}
</script>
