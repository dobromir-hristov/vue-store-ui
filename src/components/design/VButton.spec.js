import VButton from './VButton'

/**
 * @param overrides
 * @returns {{config, wrapper: Wrapper<V> | Wrapper<Vue> | *}}
 */
const createWrapper = (overrides) => {
  let config = _merge(createComponentMocks(),
    {
      slots: { default: 'BUTTON TEXT' }
    },
    overrides
  )
  const wrapper = mount(VButton, config)
  return {
    config,
    wrapper
  }
}

describe('VButton', () => {
  it('renders the VButton with default class', () => {
    const { wrapper } = createWrapper()
    expect(wrapper.classes()).toContain('v-button--primary')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the button with all props', () => {
    const { wrapper } = createWrapper({
      propsData: {
        color: 'secondary',
        outlined: true,
        size: 'lg',
        square: true,
        active: true,
        loading: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
