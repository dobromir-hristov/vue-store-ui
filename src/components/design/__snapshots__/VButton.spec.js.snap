// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VButton renders the VButton with default class 1`] = `<button class="v-button v-button--primary"><span class="v-button__text">BUTTON TEXT</span></button>`;

exports[`VButton renders the button with all props 1`] = `<button class="v-button v-button--secondary is-active is-outlined v-button--size-lg is-loading is-square"><span class="v-button__text">BUTTON TEXT</span></button>`;
